utils/auto_annotation/run_model.py

```diff
@@ -59,18 +62,63 @@ def main():
     with open(mapping_file) as json_file:
         mapping = json.load(json_file)
 
+    try:
+        mapping = mapping['label_map']
+    except KeyError:
+        print("JSON Mapping file must contain key `label_map`!")
+        print("Exiting")
+        return
+
+    mapping = {int(k): v for k, v in mapping.items()}
+
     restricted = kwargs['restricted']
     image_files = kwargs.get('image_files')
-    print(image_files, kwargs.keys())
+
+    print(image_files)
 
     if image_files:
         image_data = [cv2.imread(f) for f in image_files]
     else:
         test_image = np.ones((1024, 1980, 3), np.uint8) * 255
         image_data = [test_image,]
     attribute_spec = {}
-    results = run_inference_engine_annotation(image_data, xml_file, bin_file,mapping, attribute_spec, py_file, restricted=restricted)
-    print('Program Worked!')
+
+    results = run_inference_engine_annotation(image_data,
+                                              xml_file,
+                                              bin_file,
+                                              mapping,
+                                              attribute_spec,
+                                              py_file,
+                                              restricted=restricted)
+
+    print('Program didn\'t have any errors.')
+    show_images = kwargs.get('show_images', False)
+
+    if show_images:
+        if image_files is None:
+            print("Warning, no images provided!")
+            print('Exiting without presenting results')
+            return
+
+        if not results['shapes']:
+            print(results)
+            print("No objects detected!")
+            return
+
+        for index, data in enumerate(image_data):
+            for detection in results['shapes']:
+                points = detection['points']
+                # Cv2 doesn't like floats for drawing
+                points = [int(p) for p in points]
+                color = random_color()
+                if detection['type'] == 'rectangle':
+                    cv2.rectangle(data, (points[0], points[1]), (points[2], points[3]), color, 3)
+                elif detection['type'] == 'polygon':
+                    cv2.polylines(data, points, True, color, 3)
+                elif detection['type'] == 'polyline':
+                    cv2.polylines(data, points, True, color, 3)
+            cv2.imshow(str(index), data)
+            cv2.waitKey(0)
```

Comment:
@benhoff , better to have a parameter here (--show-images-delay with default value 0).